test/unit/math/fwd/mat/prob/ordered_logistic_test.cpp

```diff
@@ -0,0 +1,394 @@
+#include <stan/math/fwd/mat.hpp>
+#include <gtest/gtest.h>
+#include <vector>
+
+TEST(ProbDistributionsOrdLog, fd_fd) {
+  using stan::math::fvar;
+  using stan::math::ordered_logistic_lpmf;
+  using stan::math::vector_d;
+  using stan::math::vector_fd;
+  using stan::math::vector_ffd;
+
+  int y = 1;
+
+  fvar<double> lam_fd = -1.32;
+  lam_fd.d_ = 1.0;
+
+  vector_fd c_fd(3);
+  c_fd << -0.95, -0.10, 0.95;
+  for (int i = 0; i < 3; i++)
+    c_fd[i].d_ = 1.0;
```

Comment:
I'm scared to ask this since you clearly put so much work into testing the forward mode autodiff, but do you have eqs. for this stuff? I'm trying to think about how to spot check these numbers, and it's really hard to figure out the gradients.

At this comment, it's just a directional derivative. That's not so bad, but when it gets to the ffd, and then in the mixed tests the ffvs, it's really confusing what is being computed and tested. Any method to the madness that'll help me review it?

Do you have this stuff in a Mathematica notebook or anything?

Reply:
No fear necessary! I've added the math as doxygen. I've just been working off a spreadsheet for this one, I've made it public here: https://cloudstor.aarnet.edu.au/plus/s/EYqh0b6BBLQCL6Q/download

It's a bit rough, so let me know if you need anything clarified

Reply:
Is the higher order derivative stuff in here? Or is this just first derivs?